webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cli/repl/ReplScenariosSelector.groovy

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 webtau maintainers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.testingisdocumenting.webtau.cli.repl
+
+
+import org.testingisdocumenting.webtau.runner.standalone.StandaloneTest
+
+import java.util.regex.Pattern
+
+class ReplScenariosSelector {
+    private final List<StandaloneTest> availableScenarios
+    private final List<SelectInputAndResult> inputAndResults
+
+    ReplScenariosSelector(List<StandaloneTest> availableScenarios, Object... selectors) {
+        this.availableScenarios = availableScenarios
+        this.inputAndResults = selectors.collect { selectScenario(it) }.flatten()
+    }
+
+    List<SelectInputAndResult> getInputAndResults() {
+        return inputAndResults
+    }
+
+    private List<SelectInputAndResult> selectScenario(Object selector) {
+        if (selector instanceof String) {
+            return selectSingleScenarioByRegexp(selector)
+        }
+
+        if (selector instanceof Integer) {
+            return selectSingleScenarioByIdx(selector)
+        }
+
+        if (selector instanceof Map) {
+            return selectSingleScenariosByRange(selector)
+        }
+
+        return [new SelectInputAndResult(selector, -1, null)]
+    }
+    
+    private List<SelectInputAndResult> selectSingleScenarioByRegexp(String regexp) {
```

Comment:
Why do all the `selectSingle...` methods return a list? That seems contradictory to the whole single thing :)